File: content/1-basics/5-while-loop/ts/whileLoop.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

console.log("Enter numbers (enter 0 to stop):");

function askForNumber() {
    rl.question('Enter a number: ', (input) => {
        const number = parseInt(input, 10);

        if (number === 0) {
            console.log("Loop terminated because you entered 0.");
            rl.close();
        } else {
            console.log("You entered: " + number);
            askForNumber();
        }
    });
}

askForNumber();

File: content/1-basics/5-while-loop/cs/WhileLoop.cs
using System;

class Program
{
    static void Main()
    {
        int number;

        Console.WriteLine("Enter numbers (enter 0 to stop):");

        while (true)
        {
            Console.Write("Enter a number: ");
            number = int.Parse(Console.ReadLine());

            if (number == 0)
            {
                break;
            }

            Console.WriteLine("You entered: " + number);
        }

        Console.WriteLine("Loop terminated because you entered 0.");
    }
}